### PR TITLE
Update MUI5 Library.

### PR DESCRIPTION
**Pull request by yogeshmahajan-1903**
Remove scss files and its references from multiple modules. Remove scss bundling in webpacker.
Update Datefns,Dateio.Removed popper.js,sass-loader.

### CODE REVIEW

File: web/pgadmin/static/js/components/Buttons.jsx [hunk 1]
```diff
@@ -44,14 +45,19 @@ const useStyles = makeStyles((theme)=>({
   iconButton: {
     minWidth: 0,
     padding: '2px 4px',
-    '&.MuiButton-sizeSmall, &.MuiButton-outlinedSizeSmall, &.MuiButton-containedSizeSmall': {
-      padding: '1px 4px',
+    '&.MuiButton-sizeSmall, &.MuiButton-outlined.MuiButton-sizeSmall, &.MuiButton-contained.MuiButton-sizeSmall': {
+      //padding: '1px 4px'
```
Comment (adityatoshniwal) [inline web/pgadmin/static/js/components/Buttons.jsx:49]: remove commented.

File: web/pgadmin/static/js/components/Buttons.jsx [hunk 2]
```diff
@@ -84,7 +91,7 @@ const useStyles = makeStyles((theme)=>({
   },
   noBorder: {
     border: 0,
-    backgroundColor: 'transparent',
+    backgroundColor: '#00000000',
```
Comment (adityatoshniwal) [inline web/pgadmin/static/js/components/Buttons.jsx:94]: removed transparent?

File: web/pgadmin/templates/base.html [hunk 3]
```diff
@@ -84,6 +82,60 @@
 
 </head>
 <body>
+<style>
+    :root {
```
Comment (adityatoshniwal) [inline web/pgadmin/templates/base.html:85]: this should not be hard coded. It should be set based on theme.

File: web/pgadmin/templates/base.html [hunk 4]
```diff
@@ -84,6 +82,60 @@
...
+input:-webkit-autofill:hover,
+input:-webkit-autofill:focus,
+textarea:-webkit-autofill,
+textarea:-webkit-autofill:hover,
+textarea:-webkit-autofill:focus,
+select:-webkit-autofill,
+select:-webkit-autofill:hover,
+select:-webkit-autofill:focus {
+  -webkit-text-fill-color: #326690;
+  -webkit-box-shadow: 0 0 0px 1000px #d6effc inset;
```
Comment (adityatoshniwal) [inline web/pgadmin/templates/base.html:134]: even this should be based on theme.

File: web/pgadmin/tools/schema_diff/static/js/components/SchemaDiffButtonComponent.jsx [hunk 5]
```diff
@@ -169,7 +169,7 @@ export function SchemaDiffButtonComponent({ sourceData, targetData, selectedRowI
         <PgButtonGroup size="small" disabled={isDisableCompare}>
           <PrimaryButton startIcon={<CompareArrowsRoundedIcon />}
             onClick={compareDiff}>{gettext('Compare')}</PrimaryButton>
-          <PgIconButton title={gettext('Compare')} icon={<KeyboardArrowDownIcon />} color={'primary'} splitButton
+          <PgIconButton title={isDisableCompare? '':gettext('Compare')} icon={<KeyboardArrowDownIcon />} color={'primary'} splitButton
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/schema_diff/static/js/components/SchemaDiffButtonComponent.jsx:172]: why is this change needed.

File: web/regression/feature_tests/keyboard_shortcut_test.py [hunk 6]
```diff
@@ -129,9 +129,9 @@ def _update_preferences(self):
             file_menu.click()
             file_menu.send_keys(key)
 
-        maximize_button = self.page.find_by_xpath(
-            NavMenuLocators.maximize_pref_dialogue_css)
-        maximize_button.click()
+        # maximize_button = self.page.find_by_css_selector(
+        #     NavMenuLocators.maximize_pref_dialogue_css)
+        # maximize_button.click()
```
Comment (adityatoshniwal) [inline web/regression/feature_tests/keyboard_shortcut_test.py:134]: can remove commented code.

File: web/regression/feature_utils/pgadmin_page.py [hunk 7]
```diff
@@ -909,9 +923,9 @@ def find_codemirror(driver):
                 WebDriverWait(self.driver, 10).until(
                     EC.frame_to_be_available_and_switch_to_it(
                         (By.TAG_NAME, "iframe")))
-                self.find_by_css_selector(
-                    "div.dock-tab-btn[id$=\"id-query\"]").click()
-                # self.find_by_xpath("//div[text()='Query Editor']").click()
+                # self.find_by_css_selector(
+                #     "div.dock-tab-btn[id$='id-query']").click()
```
Comment (adityatoshniwal) [inline web/regression/feature_utils/pgadmin_page.py:927]: cann remove commented code.

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]
I've committed some changes related to webpack for more cleanup(It might fix some issues). Below are other issues I found.
Screenshot 2024-03-28 at 12 58 25
Screenshot 2024-03-28 at 13 00 23
Server Configuration
The underline color
General Definition Security

File: web/regression/feature_utils/pgadmin_page.py (hunk 7 above)
Comment (yogeshmahajan-1903) [reply web/regression/feature_utils/pgadmin_page.py:927]: Done

File: web/regression/feature_tests/keyboard_shortcut_test.py (hunk 6 above)
Comment (yogeshmahajan-1903) [reply web/regression/feature_tests/keyboard_shortcut_test.py:134]: Done

File: web/pgadmin/static/js/components/Buttons.jsx (hunk 1 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/js/components/Buttons.jsx:49]: Done

File: web/pgadmin/static/js/Theme/overrides/pgadmin.classes.override.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+    '.dashboard-pg-doc': {
+      backgroundImage: theme.otherVars.dashboardPgDoc,
+      display: 'inline-block',
+      width: '50px',
+      height: '50px',
+      backgroundAize: '50px 50px'
+    }
+
+
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/js/Theme/overrides/pgadmin.classes.override.js:21]: Done

File: web/pgadmin/browser/__init__.py [hunk 9]
```diff
@@ -474,10 +474,14 @@ def utils():
...
 
     prefs = Preferences.module('paths')
-
     pg_help_path_pref = prefs.preference('pg_help_path')
     pg_help_path = pg_help_path_pref.get()
 
+    # Added to have theme value available at app start page loading
+    prefs = Preferences.module('misc')
+    theme_path_pref = prefs.preference('theme')
+    theme = theme_path_pref.get()
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/browser/__init__.py:483]: Done

File: web/pgadmin/misc/themes/__init__.py [hunk 10]
```diff
@@ -5,46 +5,28 @@
...
-    theme_file_path = os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '../../static/js/generated',
-        'pgadmin.themes.json'
-    )
+    # theme_file_path = os.path.join(
+    #     os.path.dirname(os.path.realpath(__file__)),
+    #     '../../static/js/generated',
+    #     'pgadmin.themes.json'
+    # )
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/misc/themes/__init__.py:12]: Done

File: web/pgadmin/authenticate/oauth2.py [hunk 11]
```diff
@@ -134,6 +134,9 @@ def validate(self, form):
 
     def login(self, form):
         profile = self.get_user_profile()
+        current_app.logger.exception('#####')
+        current_app.logger.exception(profile)
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/authenticate/oauth2.py:138]: Yes. Done

File: web/package.json [hunk 12]
```diff
@@ -77,20 +74,21 @@
...
+    "@mui/material": "^5.15.10",
+    "@mui/styles": "^5.15.10",
+    "@mui/x-date-pickers": "^6.19.7",
     "@projectstorm/react-diagrams": "^6.6.1",
     "@simonwep/pickr": "^1.5.1",
     "@szhsin/react-menu": "^2.2.0",
     "@types/classnames": "^2.2.6",
-    "@types/react": "^16.7.18",
-    "@types/react-dom": "^17.0.11",
+    "@types/react": "^18.2.56",
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/package.json:90]: Done

File: web/pgadmin/templates/base.html (hunk 4 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/templates/base.html:134]: Done.

File: web/pgadmin/static/js/Theme/high_contrast.js [hunk 13]
```diff
@@ -130,6 +131,14 @@ export default function(basicSettings) {
         guttersBg: '#2d3a48',
         guttersFg: '#8b9cac',
       },
+      tree: {
+        treeTextFg: '#ffffff',
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/js/Theme/high_contrast.js:135]: Done.

File: web/pgadmin/static/js/components/Buttons.jsx (hunk 2 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/js/components/Buttons.jsx:94]: Done

File: web/pgadmin/misc/themes/__init__.py [hunk 14]
```diff
@@ -5,46 +5,28 @@
...
     all_themes = {
         "standard": {
             "disp_name": "Standard",
             "cssfile": "pgadmin",
             "preview_img": "standard_preview.png"
-        }
+        },
+        "dark": {
+            "disp_name": "dark",
+            "cssfile": "pgadmin.theme.dark",
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/misc/themes/__init__.py:22]: Done

File: web/pgadmin/static/bundle/app.js [hunk 15]
```diff
@@ -43,15 +43,17 @@ define('app', [
...
   // Add menus from back end.
   pgAdmin.Browser.utils.addBackendMenus(pgAdmin.Browser);
 
   // Create menus after all modules are initialized.
   MainMenuFactory.createMainMenus();
+  let theme = pgAdmin.Browser.utils.theme;
 
   ReactDOM.render(
-    <Theme>
+    <Theme theme={theme}>
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/bundle/app.js:56]: Done.

File: web/pgadmin/tools/schema_diff/static/js/components/SchemaDiffButtonComponent.jsx (hunk 5 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/tools/schema_diff/static/js/components/SchemaDiffButtonComponent.jsx:172]: Done

File: web/pgadmin/static/bundle/app.js [hunk 16]
```diff
@@ -43,15 +43,17 @@ define('app', [
   initializeModules(pgAdmin);
   initializeModules(pgAdmin.Browser);
   initializeModules(pgAdmin.Tools);
+  initializeModules(pgAdmin.Preferences);
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/bundle/app.js:46]: This is required to fetch theme data for first time.

[the remark this answers is not in the document]
Comment (adityatoshniwal) [reply web/pgadmin/static/bundle/app.js:46]: But you're using pgAdmin.utils for initial theme info.

File: web/pgadmin/static/js/Theme/overrides/reactaspen.override.js [hunk 17]
```diff
@@ -0,0 +1,293 @@
...
+      '&.red': {
+        filter: 'saturate(.5)',
+
+        '.file-icon:after': {
+          content: '""',
+          height: '8px',
+          width: '8px',
+          fontWeight: 'bold',
+          display: 'inline-block',
+          background: '#da2d38',
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/static/js/Theme/overrides/reactaspen.override.js:120]: Done

**Pull request merged**